# DP: use GNATBIND and GNATMAKE passed from the toplevel Makefile

this patch is incomplete. It still fails when building libada.

--- a/src/gcc/ada/Make-generated.in
+++ b/src/gcc/ada/Make-generated.in
@@ -18,7 +18,7 @@ GEN_IL_FLAGS = -gnata -gnat2012 -gnatw.g -gnatyg -gnatU $(GEN_IL_INCLUDES)
 ada/seinfo_tables.ads ada/seinfo_tables.adb ada/sinfo.h ada/einfo.h ada/nmake.ads ada/nmake.adb ada/seinfo.ads ada/sinfo-nodes.ads ada/sinfo-nodes.adb ada/einfo-entities.ads ada/einfo-entities.adb: ada/stamp-gen_il ; @true
 ada/stamp-gen_il: $(fsrcdir)/ada/gen_il*
 	$(MKDIR) ada/gen_il
-	cd ada/gen_il; gnatmake -q -g $(GEN_IL_FLAGS) gen_il-main
+	cd ada/gen_il; $(GNATMAKE) -q -g $(GEN_IL_FLAGS) gen_il-main
 	# Ignore errors to work around finalization issues in older compilers
 	- cd ada/gen_il; ./gen_il-main
 	$(fsrcdir)/../move-if-change ada/gen_il/seinfo_tables.ads ada/seinfo_tables.ads
@@ -39,14 +39,14 @@ ada/stamp-gen_il: $(fsrcdir)/ada/gen_il*
 # would cause bootstrapping with older compilers to fail. You can call it by
 # hand, as a sanity check that these files are legal.
 ada/seinfo_tables.o: ada/seinfo_tables.ads ada/seinfo_tables.adb
-	cd ada ; gnatmake $(GEN_IL_INCLUDES) seinfo_tables.adb -gnatU -gnatX
+	cd ada ; $(GNATMAKE) $(GEN_IL_INCLUDES) seinfo_tables.adb -gnatU -gnatX
 
 ada/snames.h ada/snames.ads ada/snames.adb : ada/stamp-snames ; @true
 ada/stamp-snames : ada/snames.ads-tmpl ada/snames.adb-tmpl ada/snames.h-tmpl ada/xsnamest.adb ada/xutil.ads ada/xutil.adb
 	-$(MKDIR) ada/bldtools/snamest
 	$(RM) $(addprefix ada/bldtools/snamest/,$(notdir $^))
 	$(CP) $^ ada/bldtools/snamest
-	cd ada/bldtools/snamest && gnatmake -q xsnamest && ./xsnamest
+	cd ada/bldtools/snamest && $(GNATMAKE) -q xsnamest && ./xsnamest
 	$(fsrcdir)/../move-if-change ada/bldtools/snamest/snames.ns ada/snames.ads
 	$(fsrcdir)/../move-if-change ada/bldtools/snamest/snames.nb ada/snames.adb
 	$(fsrcdir)/../move-if-change ada/bldtools/snamest/snames.nh ada/snames.h
